doc/api/deprecations.md

```diff
@@ -0,0 +1,567 @@
+# Deprecated APIs
+
+Node.js will deprecate APIs when either: (a) use of the API is considered to be
```

Comment:
nit: I’d phrase it as `may deprecate`, otherwise it sounds like all APIs considered “unsafe” will eventually be deprecated, and we know that’s something we may not be able to guarantee.